lib/isexpired.c

```diff
@@ -72,7 +90,7 @@ int isexpired (const struct passwd *pw, /*@null@*/const struct spwd *sp)
 	if (   (sp->sp_lstchg > 0)
 	    && (sp->sp_max >= 0)
 	    && (sp->sp_inact >= 0)
-	    && (now >= (sp->sp_lstchg + sp->sp_max + sp->sp_inact))) {
+	    && (now >= sum(sp->sp_lstchg, sp->sp_max, sp->sp_inact))) {
```

Comment:
I think I would prefer to reject overflows reporting an error, instead of silently truncating.

How about using something like `if (__builtin_add_overflow(...)) { goto error; }`?

<https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html#index-_005f_005fbuiltin_005fadd_005foverflow>,

Reply:
The isexpired function would have to be restructured a bit to actually return such error values. I'm not sure if it's worth it, considering the fact that we talk about millions of years before this is a real world scenario issue -- instead of an issue with broken files.

Reply:
Hmm.

What's the behavior now with broken files or with huge dates, and what is the behavior after your patch?  Does the behavior change?

Reply:
If the overflow occurs, `now` is larger than this limit, at which an account shall be locked. At least if signed overflows just wraps around, as it normally should do.

So, right now accounts are locked too early. With this patch, they are locked "too early" in a few million years. So, I guess that's acceptable and resembles as close as possible what a system administrator expressed with such huge values.

Reply:
LGTM.